src/ivq.tsx

```diff
@@ -12,19 +14,53 @@ export class Ivq extends KalturaPlayer.core.BasePlugin {
   constructor(name: string, player: any, config: IvqConfig) {
     super(name, player, config);
     this._player = player;
+    this._readyPromise = this._makeReadyPromise();
   }
 
   get ready() {
-    return Promise.resolve();
+    return this._readyPromise;
   }
 
-  loadMedia(): void {}
+  loadMedia(): void {
+    this._getQuiz();
+  }
 
   static isValid(): boolean {
     return true;
   }
 
-  reset(): void {}
+  reset(): void {
+    this._readyPromise = this._makeReadyPromise();
+  }
 
   destroy(): void {}
+
+  private _makeReadyPromise = () => {
+    return new Promise<void>(res => {
+      this._resolveReadyPromise = res;
+    });
+  };
+
+  private _getQuiz() {
+    this._player.provider
+    .doRequest([{loader: QuizLoader, params: {entryId: this._player.sources.id}}])
+    .then((data: Map<string, any>) => {
+      if (data && data.has(QuizLoader.id)) {
+        const quizLoader = data.get(QuizLoader.id);
+        const userEntry = quizLoader?.response?.userEntries[0];
+        if (userEntry) {
+          console.log("> userEntry", userEntry) // TODO: use userEntry data
```

Comment:
use a logger

Reply:
it's temporary code. I don't think we should add logger here and then remove it in next PR.